tfjs-backend-webgpu/src/kernels/BatchMatMul_impl.ts

```diff
@@ -97,17 +98,25 @@ export function batchMatMulImpl({
 
   const useVec4 = a.shape[2] % 4 === 0 && b.shape[2] % 4 === 0 && !transposeA &&
       !transposeB && outerShapeB >= 32;
-  let program: MatMulPackedProgram|MatMulPackedVec4Program;
+  let program: MatMulPackedProgram|MatMulPackedVec4Program
+      |MatMulSmallOutputSizeProgram;
   let dimensions = null;
-  if (useVec4) {
+  if (!transposeA && !transposeB && ((a.shape[1] <= 16 &&
+      (b.shape[2] <= 512 || b.shape[1] >= 2 * b.shape[2])) ||
+      (b.shape[2] <= 16 &&
+      (a.shape[1] <= 512 || a.shape[2] >= 2 * a.shape[1])))) {
```

Comment:
Please add an annotation why we choose ` (b.shape[2] <= 512 || b.shape[1] >= 2 * b.shape[2])` as the condition.